### PR TITLE
Docs: Fix Build & HTML5

### PR DESCRIPTION
**Pull request by ax3l**
Fix RTD build and HTML5 issues in rendering.

Add a few Sphinx warnings (duplicate labels, ill-formed commands, outdated configs, etc.) on the way.

### CODE REVIEW

File: docs/source/install/profile.rst [hunk 1]
```diff
@@ -132,11 +132,11 @@ Queue: k80 (8x NVIDIA K80 12GB)
 Summit (ORNL)
 -------------
 
-**System overview:** `link <https://www.olcf.ornl.gov/olcf-resources/compute-systems/summit/>`_
+**System overview:** `link <https://www.olcf.ornl.gov/olcf-resources/compute-systems/summit/>`__
```
Comment (ax3l) [inline docs/source/install/profile.rst:135]: Double underscores for links in RST create unnamed, in-place references. That avoids, that you get 10s of "the reference named `link` was already defined" warnings.

File: docs/source/_static/hide_chapter_titles.css [hunk 2]
```diff
@@ -1,24 +1,24 @@
 /* front page: hide chapter titles
  * needed for consistent HTML-PDF-EPUB chapters
  */
-#how-to-read-this-document.section div.section {
-    display:none;
+section#how-to-read-this-document div.section {
```
Comment (ax3l) [inline docs/source/_static/hide_chapter_titles.css:4]: This is the HTML5 update from newer sphinx and docutils versions.

File: docs/source/testing/testbuilding.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
-.. _testing-usage:
+.. _testing-new:
```
Comment (ax3l) [inline docs/source/testing/testbuilding.rst:1]: Duplicate anchor label used elsewhere already. Renamed here.

**Comment by BrianMarre** [comment]
@psychocoderHPC CI crashed with, could not connect to docker container

**Review by BrianMarre — COMMENTED** [review]
looks good to me, an you actually read the description of my example ;)

**Comment by psychocoderHPC** [comment]
@ax3l Thanks for fixing these things :heart: :heart: 

I restarted the failing tests and set the label that the CI is not required to run because these changes do not affect the code.

**Review by psychocoderHPC — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
@ax3l Could you point to a source from where you aggregated all the knowledge about these changes in Sphinx?

**Comment by ax3l** [comment]
I am humbled by the question because I fixed this just by looking at the build result, the build warnings and using my experience. That's why I added a few inline comments to the PR.

Otherwise, I usually look at the doc-linked rst primer and sphinx docs for introductories and detail questions.

@BrianMarre sure! Actually, when you compile sphinx locally there are a few more warnings visible about your parts of the docs, if you have the time to check them out.

**Comment by BrianMarre** [comment]
is on my to do list after the version 2 pull request of FLYonPIC

**Pull request merged**